functions/index.js

```diff
@@ -275,12 +312,33 @@ exports.setDefaultRank = functions.database.ref('/rank/{competitionid}/{userid}'
 
 });
 
-exports.removeDefaultRank = functions.database.ref('/rank/{competitionid}/{userid}').onDelete((snap, context) => {
+exports.removeDefaultRank = functions.database.ref('/rank/{competitionid}/{userid}/username').onDelete((snap, context) => {
+  const competitionid = context.params.competitionid;
+  const userid = context.params.userid;
+
   const snapshot = snap;
   const user = snap.val();
-  const path = snapshot._path.replace('/rank/','');
+  let path = snapshot._path.replace('/rank/','');
+  path = path.replace('/username','');
   console.log(path);
-  console.log(path + "/" + snapshot._data + " got deleted.")
-  
+
+  admin.database().ref('/rank/' + path + '/').remove();
+  admin.database().ref('/competition/' + competitionid + '/users/' + userid).remove();
+  admin.database().ref('/users/' + userid + '/usernames/' + competitionid).remove();
+  admin.database().ref('/users/' + userid + '/competitions/' + competitionid).remove();
+
+  console.log(path + "/" + snapshot._data + " got deleted. Server removed his rank.")
+
+  if (userid.length > 28) {
```

Comment:
kun je niet beter controleren op null of empty?

Reply:
Op dit moment geen makkelijke, betere manier voor maar, ik snap je punt.